src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/foundation/KotlinProcessor.kt

```diff
@@ -0,0 +1,76 @@
+package io.gitlab.arturbosch.detekt.sonar.foundation
+
+import io.gitlab.arturbosch.detekt.api.FileProcessListener
+import io.gitlab.arturbosch.detekt.core.DetektResult
+import io.gitlab.arturbosch.detekt.core.FileProcessorLocator
+import io.gitlab.arturbosch.detekt.core.processors.COMPLEXITY_KEY
+import io.gitlab.arturbosch.detekt.core.processors.LLOC_KEY
+import io.gitlab.arturbosch.detekt.core.processors.NUMBER_OF_COMMENT_LINES_KEY
+import io.gitlab.arturbosch.detekt.core.processors.ProjectCLOCProcessor
+import io.gitlab.arturbosch.detekt.core.processors.ProjectComplexityProcessor
+import io.gitlab.arturbosch.detekt.core.processors.ProjectLLOCProcessor
+import org.jetbrains.kotlin.com.intellij.openapi.util.Key
+import org.jetbrains.kotlin.psi.KtFile
+import org.sonar.api.batch.fs.InputComponent
+import org.sonar.api.batch.sensor.SensorContext
+import org.sonar.api.measures.CoreMetrics.COMMENT_LINES
+import org.sonar.api.measures.CoreMetrics.COMPLEXITY
+import org.sonar.api.measures.CoreMetrics.NCLOC
+import org.sonar.api.measures.Metric
+
+/**
+ * Class responsible for processing individual [KtFile]s and extracting useful metrics.
+ */
+class KotlinProcessor(
+		private val context: SensorContext,
+		private val fileProcessorLocator: FileProcessorLocator
+) {
+
+	private lateinit var processors: List<FileProcessListener>
+	private lateinit var llocProcessor: ProjectLLOCProcessor
+	private lateinit var clocProcessor: ProjectCLOCProcessor
+	private lateinit var complexityProcessor: ProjectComplexityProcessor
+
+	/**
+	 * Loads the processors, this method is expensive.
+	 */
+	fun load(): KotlinProcessor {
+		processors = fileProcessorLocator.load()
+		llocProcessor = processors.first { it.id == ProjectLLOCProcessor::class.java.simpleName }
+				as ProjectLLOCProcessor
+		clocProcessor = processors.first { it.id == ProjectCLOCProcessor::class.java.simpleName }
+				as ProjectCLOCProcessor
+		complexityProcessor = processors.first { it.id == ProjectComplexityProcessor::class.java.simpleName }
```

Comment:
Just load the processors with an init block and store them all in one variable, we do not need the specific ones. See comment later :)